src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -628,17 +632,28 @@ public String getMailAddressAttributeName() {
         return StringUtils.defaultString(mailAddressAttributeName, DescriptorImpl.DEFAULT_MAILADDRESS_ATTRIBUTE_NAME);
     }
 
-    public SecurityComponents createSecurityComponents() {
+    /**
+     * Creates security components.
+     * @return Created {@link SecurityComponents}
+     * @throws Error Execution error
+     */
+    @Override
+    public @Nonnull SecurityComponents createSecurityComponents() {
```

Comment:
When annotating the method, please annotate the method. Just because some annotations have semantic meaning on the return type does not mean you should pretend you are annotating the return type. You are annotating the method.